### ISSUE TITLE
Warden not working on shared hosting - open_basedir restriction in effect. open_basedir restriction in effect. File(/dev/urandom) is not within the allowed path(s):

### ISSUE THREAD

**Issue by PrimozRome** [dre1080/warden#52]
Warden does not work on MeditaTemple's shared grid hsoting. The error is:

Login failed! is_readable() [function.is-readable]: open_basedir restriction in effect. File(/dev/urandom) is not within the allowed path(s): (/nfs:/tmp:/usr/local:/etc/apache2/gs-bin)

is_readable('/dev/urandom') is called by class PasswordHash (phpass) ...

**Comment by PrimozRome** [comment]
This is actually PHPass problem, not Warden's but anyway it is good to know... putting @ before is_readable('/dev/urandom') in PasswordHash.php solves the problem...

```
function get_random_bytes($count)
{
    $output = '';
    if (@is_readable('/dev/urandom') &&
        ($fh = @fopen('/dev/urandom', 'rb'))) {
        $output = fread($fh, $count);
        fclose($fh);
    }

    if (strlen($output) < $count) {
        $output = '';
        for ($i = 0; $i < $count; $i += 16) {
            $this->random_state =
                md5(microtime() . $this->random_state);
            $output .=
                pack('H*', md5($this->random_state));
        }
        $output = substr($output, 0, $count);
    }

    return $output;
}
```

**Comment by dre1080** [comment]
Closing as this was fixed by noxify@b485b9ece675cfeaae91b547bfb9527fb1c9293c
